drivers/soundwire/bus.c

```diff
@@ -21,6 +21,17 @@ int sdw_add_bus_master(struct sdw_bus *bus)
 	struct sdw_master_prop *prop = NULL;
 	int ret;
 
+	if (!bus) {
```

Comment:
> why this wasn't needed before and is needed now? Are we adding call paths, where this function can be called with `bus == NULL`?

Hmm, this is a valid point. I will remove the bus test in the PR. Thanks @lyakh

Reply:
I would like to keep this test on bus being valid.
The bus is allocated somewhere else, and in the initial version the bus was allocated as part of another context which was already tested.
Now that we are making the bus the primary handle for all master configurations, we have to test that it was allocated properly in bus_master_add/delete/startup/process_wake.

It's just basic defensive programming when you don't control how the bus is allocated to avoid NULL pointer dereferences.